ingestion/src/metadata/ingestion/source/pipeline/openlineage/metadata.py

```diff
@@ -0,0 +1,534 @@
+                    columns=columns,
+                    databaseSchema=om_schema_fqn,
+                )
+
+                return Either(right=request)
+        else:
+            pass
+
+        return None
+
+    @classmethod
+    def _get_ol_table_name(cls, table: Dict) -> str:
+        return "/".join(table.get(f) for f in ["namespace", "name"]).replace("//", "/")
+
+    def _build_ol_name_to_fqn_map(self, tables: List):
+        result = {}
+
+        for table in tables:
+            table_fqn = self._get_table_fqn(OpenlineageSource._get_table_details(table))
+
+            if table_fqn:
+                result[OpenlineageSource._get_ol_table_name(table)] = table_fqn
+
+        return result
+
+    @classmethod
+    def _create_output_lineage_dict(
+        cls, lineage_info: List[Tuple[str, str, str, str]]
+    ) -> Dict[str, Dict[str, List[ColumnLineage]]]:
+        result = defaultdict(lambda: defaultdict(list))
+        for (output_table, input_table, output_column), group in groupby(
+            lineage_info, lambda x: x[:3]
+        ):
+            input_columns = [input_col for _, _, _, input_col in group]
+
+            result[output_table][input_table] += [
+                ColumnLineage(toColumn=output_column, fromColumns=input_columns)
+            ]
+
+        return result
+
+    def _get_column_lineage(
+        self, inputs: List, outputs: List
+    ) -> Dict[str, Dict[str, List[ColumnLineage]]]:
+        _result: List = []
+
+        ol_name_to_fqn_map = self._build_ol_name_to_fqn_map(inputs + outputs)
+
+        for table in outputs:
+            output_table_fqn = self._get_table_fqn(
+                OpenlineageSource._get_table_details(table)
+            )
+            for field_name, field_spec in (
+                table.get("facets", {})
+                .get("columnLineage", {})
+                .get("fields", {})
+                .items()
+            ):
+                for input_field in field_spec.get("inputFields", []):
+                    input_table_ol_name = OpenlineageSource._get_ol_table_name(
+                        input_field
+                    )
+
+                    _result.append(  # output table, input table, output column, input column
+                        (
+                            output_table_fqn,
+                            ol_name_to_fqn_map.get(input_table_ol_name),
+                            f"{output_table_fqn}.{field_name}",
+                            f'{ol_name_to_fqn_map.get(input_table_ol_name)}.{input_field.get("field")}',
+                        )
+                    )
+
+        return OpenlineageSource._create_output_lineage_dict(_result)
+
+    def yield_pipeline(
+        self, pipeline_details: OpenLineageEvent
+    ) -> Iterable[Either[CreatePipelineRequest]]:
+        pipeline_name = self.get_pipeline_name(pipeline_details)
+        try:
+            description = f"""```json
+            {json.dumps(pipeline_details.run_facet, indent=4).strip()}```"""
```

Comment:
I'm not sure if we should actually put the json in the description as we are already translating this json into lineage in OMD

Reply:
OpenLineage run facets are very dynamic and bear a lot of metadata we don't touch anywhere (apart from run facet parent job name).